package-system/sdformat/docker_build_sdformat.sh

```diff
@@ -0,0 +1,211 @@
+echo $CMD
+eval $CMD
+if [ $? -ne 0 ]
+then
+    echo "Error configuring $DEP_NAME"
+    exit 1
+fi
+
+echo "Building and installing $DEP_NAME"
+CMD="cmake --build $GZ_CMAKE_BUILD_FOLDER --target install --config Release"
+echo $CMD
+eval $CMD
+if [ $? -ne 0 ]
+then
+    echo "Error building $DEP_NAME"
+    exit 1
+fi
+
+popd
+
+# Build the dependent gz-utils library
+# NOTE: This must be done after gz-cmake it depends on that library
+DEP_NAME=gz-utils
+GZ_UTILS_SRC_FOLDER=${GIT_DEPS_BASE}/$DEP_NAME
+GZ_UTILS_BUILD_FOLDER=${GIT_DEPS_BUILD_ROOT}/$DEP_NAME
+# install gz-utils to the local filesystem
+GZ_UTILS_INSTALL_FOLDER=${LOCAL_FILESYSTEM}/deps/install
+
+# Append the gz-utils install folder
+DEP_INSTALL_PATHS+=( $GZ_UTILS_INSTALL_FOLDER )
+pushd $GZ_UTILS_SRC_FOLDER
+
+echo "Configuring $DEP_NAME"
+CMD="cmake -B ${GZ_UTILS_BUILD_FOLDER} -S. -DCMAKE_INSTALL_PREFIX=${GZ_UTILS_INSTALL_FOLDER} -DBUILD_TESTING=OFF"
+echo $CMD
+eval $CMD
+if [ $? -ne 0 ]
+then
+    echo "Error configuring $DEP_NAME"
+    exit 1
+fi
+
+echo "Building and installing $DEP_NAME"
+CMD="cmake --build $GZ_UTILS_BUILD_FOLDER --target install --config Release"
+echo $CMD
+eval $CMD
+if [ $? -ne 0 ]
+then
+    echo "Error building $DEP_NAME"
+    exit 1
+fi
+
+popd
+
+# Build the dependent gz-math library
+# NOTE: This must be done after gz-cmake and gz-utils as it depends on those libraries
+DEP_NAME=gz-math
+GZ_MATH_SRC_FOLDER=${GIT_DEPS_BASE}/$DEP_NAME
+GZ_MATH_BUILD_FOLDER=${GIT_DEPS_BUILD_ROOT}/$DEP_NAME
+GZ_MATH_INSTALL_FOLDER=${LOCAL_FILESYSTEM}/deps/install
+
+# Append the gz-math install folder
+DEP_INSTALL_PATHS+=( $GZ_MATH_INSTALL_FOLDER )
+pushd $GZ_MATH_SRC_FOLDER
+
+echo "Configuring $DEP_NAME"
+CMD="cmake -B ${GZ_MATH_BUILD_FOLDER} -S. -DCMAKE_INSTALL_PREFIX=${GZ_MATH_INSTALL_FOLDER} -DSKIP_SWIG=ON -DSKIP_PYBIND11=ON -DBUILD_TESTING=OFF"
+echo $CMD
+eval $CMD
+if [ $? -ne 0 ]
+then
+    echo "Error configuring $DEP_NAME"
+    exit 1
+fi
+
+echo "Building and installing $DEP_NAME"
+CMD="cmake --build $GZ_MATH_BUILD_FOLDER --target install --config Release"
+echo $CMD
+eval $CMD
+if [ $? -ne 0 ]
+then
+    echo "Error building $DEP_NAME"
+    exit 1
+fi
+
+popd
+
+# Now build the SDF library
+pushd ${LOCAL_FILESYSTEM}/src
+
+# Convert the dependent install path array in bash to a cmake list
+CMAKE_PREFIX_PATH=${DEP_INSTALL_PATHS[0]}
+# splace the first element of the array and then join with the semicolon CMake delimiter
+SPLICED_INSTALL_PATHS=(${DEP_INSTALL_PATHS[@]:1})
+CMAKE_PREFIX_PATH+=$(printf ";%s" "${SPLICED_INSTALL_PATHS[@]}")
+# Supply the CMAKE_PREFIX_PATH to allow the dependent libraries of gz-cmake to be located
+
+# Remove the build folder if it exist
+if [ -d ${BUILD_FOLDER} ]; then
+    rm -rf ${BUILD_FOLDER}
+fi
+
+echo "Configuring ${LIB_NAME}"
+CMD="cmake -B ${BUILD_FOLDER} -S. -DUSE_INTERNAL_URDF=ON -DBUILD_TESTING=OFF -DCMAKE_INSTALL_PREFIX=${INSTALL_FOLDER} -DCMAKE_PREFIX_PATH=\"${CMAKE_PREFIX_PATH}\""
```

Comment:
```suggestion
# When building sdformat, set it to use its internal copy of urdfdom for parsing urdf files instead of relying on an externally-installed package.
# This keeps the dependencies self-contained.
CMD="cmake -B ${BUILD_FOLDER} -S. -DUSE_INTERNAL_URDF=ON -DBUILD_TESTING=OFF -DCMAKE_INSTALL_PREFIX=${INSTALL_FOLDER} -DCMAKE_PREFIX_PATH=\"${CMAKE_PREFIX_PATH}\""
```